### DISCUSSION TITLE
Is there a way to rename the navigation buttons for a single page?

### DISCUSSION THREAD

**Discussion by storyteller-dev** [ctreffe/alfred#132] [Alfred Q&A] [answered]
I know you can rename the navigation buttons in the configuration in this section:

```
# SECTION: navigation --------------------------------------------------
# Defines the texts on alfreds navigation buttons.
# ----------------------------------------------------------------------
[navigation]
forward = Weiter                # Text on the "forward" button
backward = Zurück               # Text on the "backward" button
finish = Beenden                # Text on the "finish" button
```

But can you rename them for only a certain page?
I'd like to rename the forward button to "I agree" when the consent form of my experiment is shown.

**Comment by jobrachem** [comment, marked as answer]
There are basically two ways: 1) Use custom buttons, or 2) temporarily change the configuration. I recommend **option one**, because it is more explicit and does not involve messing around with the configuration from within the script.py, which should be avoided if possible to avoid confusion.

## Custom buttons (recommended)

There's always the possibility to hide the default navigation buttons by adding an `alfred3.HideNavigation` element to a page. You can then manually add `alfred3.BackButton` ([docs](https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.element.action.BackButton.html?highlight=backbutton)) and `alfred3.ForwardButton` ([docs](https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.element.action.ForwardButton.html?highlight=forwardbutton)) back to the page.

In the example below, I use `alfred3.VerticalSpace` and `alfred3.Hline`, and organize the buttons in a `alfred3.Row` to emulate the looks of the default navigation.

```python
import alfred3 as al

exp = al.Experiment()

exp += al.Page("Demo Page", name="demo1")

@exp.member
class Demo(al.Page):
    def on_exp_access(self):
        self += al.Text("Demo text", align="center")

        self += al.HideNavigation()
        
        back = al.BackButton("Disagree", align="right")
        forward = al.ForwardButton("Agree")
        
        self += al.VerticalSpace("300px")
        self += al.Hline()
        self += al.Row(back, forward)
```

## Temporarily change the config

The `ExperimentSession.config` and `ExperimentSession.secrets` objects are basically nothing else but slightly modified configparser objects ([docs](https://docs.python.org/3.7/library/configparser.html)). So you can, for instance, use the method `configparser.read_dict` to change the configuration on when a specific page is shown. Keep in mind though, that this will change the configuration for the complete experiment, so if you want to change the navigation only for a single page, you have to change it back to the previous value on hiding. Here's an example:

```python
import alfred3 as al

exp = al.Experiment()

exp += al.Page("Demo Page", name="demo1")

@exp.member
class Demo(al.Page):
    def on_each_show(self):
        self.exp.config.read_dict({"navigation": {"forward": "Agree"}})
    
    def on_each_hide(self):
        self.exp.config.read_dict({"navigation": {"forward": "Weiter"}})
```
